src/sync/installation.ts

```diff
@@ -314,56 +316,66 @@ async function doProcessInstallation(app, data: BackfillMessagePayload, sentry:
 		statsd.increment(metricTaskStatus.complete, [`type: ${nextTask.task}`]);
 
 	} catch (err) {
-		const rateLimit = Number(err?.headers?.["x-ratelimit-reset"]);
-		const delay = Math.max(Date.now() - rateLimit * 1000, 0);
-
-		if (delay) {
-			// if not NaN or 0
-			logger.info({ delay }, `Delaying job for ${delay}ms`);
-			scheduleNextTask(delay);
-			return;
-		}
+		await handleBackfillError(err, logger, scheduleNextTask, data, task, repositoryId, subscription, nextTask);
+	}
+}
 
-		if (String(err).includes("connect ETIMEDOUT")) {
-			// There was a network connection issue.
-			// Add the job back to the queue with a 5 second delay
-			logger.warn("ETIMEDOUT error, retrying in 5 seconds");
-			scheduleNextTask(5_000);
-			return;
-		}
+const handleBackfillError = async (err,
+	logger: LoggerWithTarget,
+	scheduleNextTask: (delayMs) => void, data: BackfillMessagePayload,
+	task: "pull" | "commit" | "branch",
+	repositoryId: string,
+	subscription: SubscriptionClass,
+	nextTask: Task) : Promise<void> => {
 
-		if (
-			String(err.message).includes(
-				"You have triggered an abuse detection mechanism"
-			)
-		) {
-			// Too much server processing time, wait 60 seconds and try again
-			logger.warn("Abuse detection triggered. Retrying in 60 seconds");
-			scheduleNextTask(60_000);
-			return;
-		}
+	const rateLimit = (err instanceof RateLimitingError || err instanceof OldRateLimitingError) ? err.rateLimitReset : Number(err?.headers?.["x-ratelimit-reset"]);
+	const delay = Math.max(rateLimit * 1000 - Date.now(), 0);
```

Comment:
A few things were changed in error handling.

Before:
```
	const rateLimit = Number(err?.headers?.["x-ratelimit-reset"]);
	const delay = Math.max(Date.now() - rateLimit * 1000, 0);
```

Delay calculation was wrong, because it would be always zero (rate-limit-reset time is always greater than the current time)

I've added error type check and getting rate limit from our rate limiting exceptions